File: 프로그래머스/lv0/120825. 문자 반복 출력하기/문자 반복 출력하기.js
function solution(my_string, n) {
    var answer = '';
 
    for(var i=0; i<my_string.length; i++){
        for(var j=0; j<=n-1; j++){
            answer += my_string.charAt(i);
            //charAt 함수: 문자를 반환
            //my_string i번째를 n(i가 0부터 이므로 n-1)만큼 반복해서 answer에 대입
        }
    }
    return answer;
}


//function solution(my_string, n) {
//    var answer = '';
//    var arr = my_string.split("");
    
//    for(var i=0; i<my_string.length; i++){
//        answer += arr[i].repeat(n);
//    }
//    return answer;
//}

File: 프로그래머스/lv0/120825. 문자 반복 출력하기/문자 반복 출력하기.java
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        String[] arr = my_string.split(""); 
        //각 문자를 n만큼 반복해야하므로 문자를 분리하여
        //String 타입의 arr 배열에 담아준다.
        
        for(int i=0; i<my_string.length(); i++){ //my_string의 한글자씩 빼온다.
            answer += arr[i].repeat(n); //repeat(반복횟수)로 담아준다.
        }
        
        
        return answer;
    }
}